cmd/podman/images/scp.go

```diff
@@ -0,0 +1,330 @@
+		var buffer bytes.Buffer
+		session.Stdout = &buffer
+		if err := session.Run(run); err != nil { // run the command declared above
+			return "", err
+		}
+		return buffer.String(), nil // return the output of image load on the machine, should be "Loaded Image(s)..."
+	} // else we are loading from the ssh client
+	podman := os.Args[0]
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v // get path to podman bin
+	}
+	run := ""
+	if tag != "" {
+		return "", errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+	}
+	run = podman + " image save " + imageFileOut + " --output=" + imageFileIn // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+
+	if err := session.Run(run); err != nil { // run the command declared above
+		return "", err
+	}
+	n, err := scpD.CopyFrom(dial, imageFileIn, imageFileIn)
+	if err != nil {
+		fmt.Println(n, " Bytes copied before error")
+		return " ", err
+	}
+	return "", nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is an array of service destinations
+func parseLocation(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	switch len(args) {
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) { // if an image is specified, this mean we are loading from our client
+				scpOpts.Connections = append(scpOpts.Connections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+				} else { // else its a load save (order of args)
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+				}
+			}
+		} else if strings.Contains(args[1], "::") && len(scpOpts.Connections) < 2 { // if we are given image host::
+			scpOpts.ToRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[1])
+		}
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[0])
+		}
+	}
```

Comment:
There should be a `default: return an error` here, to simplify reading the code.

No need to get fancy with user-targeted messages, "internal error: unexpected len(args) == %d" or something that can be fine if the caller is expected to validate `len(args)` (except that right now that doesn’t seem to happen).

---

(Personal absolutely non-blocking aesthetic preference: I’d find it more natural to have the `1` case before the `2` case.)

Reply:
fixed